src/app/external-cluster-wizard/steps/external-cluster/provider/eks/component.ts

```diff
@@ -0,0 +1,182 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Component, forwardRef, Input, OnDestroy, OnInit} from '@angular/core';
+import {
+  ControlValueAccessor,
+  FormBuilder,
+  NG_VALIDATORS,
+  NG_VALUE_ACCESSOR,
+  Validator,
+  Validators,
+} from '@angular/forms';
+import {NameGeneratorService} from '@core/services/name-generator';
+import {NodeDataService} from '@core/services/node-data/service';
+import {ExternalClusterService} from '@shared/components/add-external-cluster-dialog/steps/service';
+import {forkJoin} from 'rxjs';
+import {finalize, takeUntil} from 'rxjs/operators';
+import {StepBase} from '@app/external-cluster-wizard/base';
+
+enum Controls {
+  Name = 'name',
+  Version = 'version',
+  RoleArn = 'roleArn', // ClusterServiceRole
+  Vpcs = 'vpcs',
+  SubnetIds = 'subnetIds',
+  SecurityGroupsIds = 'securityGroupIds',
+}
+
+@Component({
+  selector: 'km-eks-external-cluster',
+  templateUrl: './template.html',
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => EKSExternalClusterComponent),
+      multi: true,
+    },
+    {
+      provide: NG_VALIDATORS,
+      useExisting: forwardRef(() => EKSExternalClusterComponent),
+      multi: true,
+    },
+  ],
+})
+export class EKSExternalClusterComponent
+  extends StepBase
+  implements OnInit, OnDestroy, ControlValueAccessor, Validator
+{
+  readonly Controls = Controls;
+  isLoading: boolean;
+  vpcs: string[] = [];
+  subnetIds: string[] = [];
+  securityGroupIds: string[] = [];
+  @Input() projectID: string;
+  @Input() showExtended = false; // Used only when in dialog mode.
+
+  constructor(
+    private readonly _builder: FormBuilder,
+    private readonly _externalClusterService: ExternalClusterService,
+    private readonly _nameGenerator: NameGeneratorService,
+    private readonly _nodeDataService: NodeDataService
+  ) {
+    super();
+  }
+
+  ngOnInit(): void {
+    this._initForm();
+    this._intiLookupValues();
+    this._initSubscriptions();
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  generateName(): void {
+    this.form.get(Controls.Name).setValue(this._nameGenerator.generateName());
+  }
+
+  reset(): void {
+    this.subnetIds = [];
+    this.securityGroupIds = [];
+  }
+
+  private _initForm() {
+    this.form = this._builder.group({
+      [Controls.Name]: this._builder.control(this._nodeDataService.nodeData.name),
+      [Controls.RoleArn]: this._builder.control('', Validators.required),
+      [Controls.Version]: this._builder.control(''),
+      [Controls.Vpcs]: this._builder.control('', Validators.required), // this will be used to show/hide subnets/security-groups dropdown
```

Comment:
```suggestion
      [Controls.Vpc]: this._builder.control(null, Validators.required), // this will be used to show/hide subnets/security-groups dropdown
```